src/Aspire.ProjectTemplates/templates/aspire-starter/AspireStarterApplication.1.Tests/WebTests.cs

```diff
@@ -0,0 +1,22 @@
+using System.Net;
+
+namespace AspireStarterApplication.1.Tests;
+
+public class WebTests
+{
+    [Fact]
+    public async Task GetWebResourceRootReturnsOkStatusCode()
+    {
+        // Arrange
+        var appHost = await DistributedApplicationTestingBuilder.CreateAsync<Projects.AspireStarterApplication.1_AppHost>();
+        await using var app = await appHost.BuildAsync();
+        await app.StartAsync();
+
+        // Act
+        var httpClient = app.CreateHttpClient("webfrontend");
+        var response = await httpClient.GetAsync("/");
+
+        // Assert
+        Assert.Equal(HttpStatusCode.OK, response.StatusCode);
```

Comment:
A nice thing we do in our tests is:

https://github.com/dotnet/aspire/blob/9073e0788b830aafa1e0dede786e861143e1a8b9/tests/Aspire.EndToEnd.Tests/IntegrationServicesTests.cs#L38-L41

That way if it fails, you get the responseContent logged.

Reply:
Because we have resiliency on the HttpClient returned from CreateHttpClient, you can't actually do it quite like that, as if the status code wasn't a success status code, it will retry until it fails anyway. I think keeping this test in the template simple is probably best for now.

Reply:
Doesn't resilience only retry on certain failures? If a service returns 503, it wouldn't retry that, would it?

Reply:
Actually I'm not sure, I thought a 500 is exactly the kind of thing it would retry, vs. a 300 or 400 status code. Obviously connection level issues would be retried too.